Adds api call to the RAR server

thomaj opened the pull request:
Now gets articles from RAR with a call to the server which gets these articles.  Then displays the retrieved articles by the category with the greatest confidence score by a simple algorithm.

This branch also contains the RAR server built by Evan

Commit:
component restructuring

evankozliner committed:
Adds a scraper and json dumper

evankozliner committed:
Adds a list of SOLR commands so we don't need to hunt down the API reference

evankozliner committed:
Added makefile for necessary directories and sqlite db. Short shell script handles scrapping RAR and replacing it

evankozliner committed:
Solr configuration, including schema

Commit:
Finished component restructoring

Commit:
Passing data test

evankozliner committed:
Adds a little flask server that handles SOLR queries

evankozliner committed:
Couple of bug fixes, should be ready for demo

Commit:
Added loading spinner

Commit:
Updated transitions between components

evankozliner committed:
Adds an upload for the writer for immediately after the text file is finished

evankozliner committed:
Updates readme and removes scraper dependency in server

evankozliner committed:
Realized I forgot to upload descriptions, will recreate later. Added published and emotions to response

Commit:
Merge remote-tracking branch 'origin/RAR-addition' into add-RAR-call

Mergining in the server for Retrieve and Rank

evankozliner committed:
Changes port

Commit:
Merge in of RAR-branch and began set call set up

Began set up for api call to the RAR backend server

evankozliner committed:
adds cors for local usage

Commit:
Mergered in fix in the python server

Commit:
Finished adding RAR call to the front end

Articles can be retrieved from the front end and displayed

evankozliner committed:
Adds the description to SOLR

Commit:
Worked on styling

Commit:
Merged in RAR server fix

Commit:
Worked on good article styling

Commit:
Added sentiment category analysis

Added javascript Sentiment module to determine which category the article blongs to and with what confidence.  Also added more styling.

Commit:
Minor fixes

Fixed jump in transition and added headers

Commit:
Actually fixed transition and changes styling

evankozliner commented on src/Sentiment.js:
```diff
@@ -0,0 +1,91 @@
+  var disgustConfidence = obj.disgust
+  var joyConfidence = obj.joy
+
+  // 'good' confidence. Just use joy confidence
```
seems reasonable

evankozliner commented on src/components/ValidQuestion.vue:
```diff
@@ -0,0 +1,65 @@
+      console.log(resp)
+      if (!resp.hasOwnProperty('solrErrorMessage')) {
+        var arrOfArticles = resp.response.docs
+        this.articles = Sentiment.sortBySentiment(arrOfArticles)
```
As it stands right now unrelated articles are displaying on the page. After we've gotten the sentiment here we should definitely take into account the order in which the articles were returned from SOLR, as it dictates the articles relevance

evankozliner reviewed (commented):
SOLR relevance still needs taken into account

evankozliner committed:
Fixes issue with short name

Commit:
Added Article component and can now sort by date

Commit:
Articles are now sorted and displayed by date

Articles are sorted by date with the most recent article at the top of the page. The color next to it represents the sentiment towards that article

evankozliner commented:
Still needs
* Date to be displayed
* Some sort of relevancy filter (probably on back end)

But we can have a "touch up" branch some time soon.
